lib/web_ui/lib/src/engine/semantics/dialog.dart

```diff
@@ -16,6 +16,39 @@ class Dialog extends PrimaryRoleManager {
     // names its own route an `aria-label` is used instead of `aria-describedby`.
     addFocusManagement();
     addLiveRegion();
+
+    // When a route/dialog shows up it is expected that the screen reader will
+    // focus on something inside it. There could be two possibilities:
+    //
+    // 1. The framework explicitly marked a node inside the dialog as focused
+    //    via the `isFocusable` and `isFocused` flags. In this case, the node
+    //    will request focus directly and there's nothing to do on top of that.
+    // 2. No node inside the route takes focus explicitly. In this case, the
```

Comment:
In case this route is reveal when the previous top-most routes are popped, you can listen to the send focus event if we start sending event for web in framework.

Reply:
Yep. I've been getting updates from @hangyujin about it. I'm planning to implementing for web, if/when it's ready on the framework side.

Reply:
it has been implemented on ios and android now. related PR: https://github.com/flutter/engine/pull/48252 https://github.com/flutter/flutter/pull/135771